lldb/source/Symbol/CoreDumpOptions.cpp

```diff
@@ -0,0 +1,37 @@
+//===-- CoreDumpOptions.cpp -------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "lldb/Symbol/CoreDumpOptions.h"
+
+using namespace lldb;
+using namespace lldb_private;
+
+void CoreDumpOptions::SetCoreDumpPluginName(const llvm::StringRef name) {
+  m_core_dump_plugin_name = name.data();
+}
+
+std::optional<llvm::StringRef> CoreDumpOptions::GetCoreDumpPluginName() const {
+  if (!m_core_dump_plugin_name)
+    return std::nullopt;
+  return m_core_dump_plugin_name->data();
```

Comment:
No need to call `data()` right? Can you not make a StringRef from a std::string? This would incur an additional `strlen`